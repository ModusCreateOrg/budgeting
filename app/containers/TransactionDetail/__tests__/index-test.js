import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { TransactionDetail } from '../index';

const mockTransaction = {
  id: 10,
  description: 'Testing transaction',
  value: -651.24,
  categoryId: 1,
};

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <TransactionDetail currentTransaction={mockTransaction} totalInflow={4521.2} totalOutflow={521.2} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without a valid transaction', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <TransactionDetail />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
