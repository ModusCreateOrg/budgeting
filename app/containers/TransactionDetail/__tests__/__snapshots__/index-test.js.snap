// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="transactionDetail"
>
  <h1>
    Testing transaction
  </h1>
  <h2
    className="neg"
  >
    -124.95%
  </h2>
  <a
    aria-current={null}
    className="navLink"
    href="/budget"
    onClick={[Function]}
  >
    Back
  </a>
</div>
`;

exports[`renders correctly without a valid transaction 1`] = `
<div
  className="transactionDetail"
>
  <h2>
    Invalid Transaction
  </h2>
  <a
    aria-current={null}
    className="navLink"
    href="/budget"
    onClick={[Function]}
  >
    Back
  </a>
</div>
`;
